app/scripts/lib/util.js

```diff
@@ -158,6 +159,45 @@ function getChainType(chainId) {
   return 'custom';
 }
 
+/**
+ * Loops through the nested allCollectibles collection and finds and returns a collectible
+ * that matches the contractAddress/tokenId pair passed as args.
+ *
+ * @param {Object} allCollectibles - The collection of all collectibles the user has ever owned organized by account and chainId.
+ * @param {string} contractAddress - The address of the contract to match against.
+ * @param {string} collectibleTokenId - The collectible tokenId to match against.
+ * @returns {Object | null} - The matching collectible if found, null if not found.
+ *
+ */
+function getKnownCollectible(
+  allCollectibles,
+  contractAddress,
+  collectibleTokenId,
+) {
+  for (const account in allCollectibles) {
+    if (Object.prototype.hasOwnProperty.call(allCollectibles, account)) {
+      for (const chainId in allCollectibles[account]) {
+        if (
+          Object.prototype.hasOwnProperty.call(
+            allCollectibles[account],
+            chainId,
+          )
+        ) {
+          const matchingCollectible = allCollectibles[account][chainId].find(
+            ({ address, tokenId }) =>
+              isEqualCaseInsensitive(address, contractAddress) &&
+              tokenId === collectibleTokenId,
+          );
+          if (matchingCollectible) {
+            return matchingCollectible;
+          }
+        }
+      }
+    }
+  }
+  return null;
```

Comment:
Cool, thanks for adding this. One final suggestion on this. I notice we have Lodash so I think we should be able to simplify this slightly:

``` suggestion
  return find(allCollectibles, (collectiblesByChainId, account) => {
    return find(collectiblesByChainId, (collectibles, chainId) => {
      return collectibles.find(({ address, tokenId }) => (
        isEqualCaseInsensitive(address, contractAddress) &&
          tokenId === collectibleTokenId,
      ));
    });
  });
```

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/13048/commits/43808729a9fa9a709fcfcd469aded490ff4f08fc